dbms/src/DataStreams/WindowBlockInputStream.cpp

```diff
@@ -273,91 +280,120 @@ Int64 WindowTransformAction::getPartitionEndRow(size_t block_rows)
     return left;
 }
 
-RowNumber WindowTransformAction::stepToFrameStart(const RowNumber & current_row, const WindowFrame & frame)
+std::tuple<RowNumber, bool> WindowTransformAction::stepToFrameStart(
```

Comment:
Please add some explanation why this is changed, comment or pr description

Reply:
> Please add some explanation why this is changed, comment or pr description

okk